packages/relic/lib/src/headers/typed/headers/cookie_header.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:relic/src/headers/extension/string_list_extensions.dart';
+import 'package:relic/src/headers/typed/base/typed_header.dart';
+import 'package:collection/collection.dart';
+
+/// A class representing the HTTP Cookie header.
+///
+/// This class manages the parsing and representation of cookies.
+class CookieHeader extends TypedHeader {
+  /// The list of cookies.
+  final List<Cookie> cookies;
+
+  /// Constructs a [CookieHeader] instance with the specified cookies.
+  const CookieHeader({required this.cookies});
+
+  /// Parses the Cookie header value and returns a [CookieHeader] instance.
+  ///
+  /// This method processes the header value, extracting the cookies into a list.
+  factory CookieHeader.parse(String value) {
+    if (value.isEmpty) {
+      throw FormatException('Value cannot be empty');
+    }
+
+    final splitValues = value.splitTrimAndFilterUnique(separator: ';');
+
+    final cookies = splitValues.map(Cookie.parse).toList();
+
+    return CookieHeader(cookies: cookies);
+  }
+
+  Cookie? getCookie(String name) {
+    return cookies.firstWhereOrNull((cookie) => cookie.name == name);
+  }
+
+  /// Converts the [CookieHeader] instance into a string representation
+  /// suitable for HTTP headers.
+  @override
+  String toHeaderString() {
+    return cookies.map((cookie) => cookie.toHeaderString()).join('; ');
+  }
+
+  @override
+  String toString() {
+    return 'CookieHeader(cookies: $cookies)';
+  }
+}
+
+/// A class representing a single cookie.
+class Cookie {
+  /// The name of the cookie.
+  final String name;
+
+  /// The value of the cookie.
+  final String value;
+
+  /// Constructs a [Cookie] instance with the specified name and value.
+  Cookie({
+    required this.name,
+    required this.value,
+  })  : assert(isNameValid(name), 'Invalid cookie name'),
+        assert(isValueValid(value), 'Invalid cookie value');
```

Comment:
@SandPod Good point, and yes, this was intentional. The goal is to ensure that invalid headers are caught as early as possible, regardless of how the cookies are constructed. Since someone could use the constructor directly to build cookies, it’s critical that the class validates inputs and throws exceptions when invalid headers are provided in any scenario.
So what do you think ?

Reply:
@SandPod you are right, this will be removed by the compiler in `release mode`, and we need to throw in `release mode` also. Thank you!